### DISCUSSION TITLE
Is there a tool for creating and updating formula with many perl deps?

### DISCUSSION THREAD

**Discussion by kencu** [Homebrew/discussions#4881] [Writing Formulae/Casks] [answered]
### Output of `brew config`

```shell
% brew config
HOMEBREW_VERSION: 4.1.18-33-g83b4b2f
ORIGIN: https://github.com/Homebrew/brew
HEAD: 83b4b2f2f4b28e1f035e88ac41140f1c8b0c9c9b
Last commit: 2 days ago
Core tap origin: https://github.com/Homebrew/homebrew-core
Core tap HEAD: c6ae22d8b04462e3c63db99da59369904954d876
Core tap last commit: 20 hours ago
Core tap branch: master
Core tap JSON: 02 Nov 05:52 UTC
HOMEBREW_PREFIX: /opt/homebrew
HOMEBREW_CASK_OPTS: []
HOMEBREW_MAKE_JOBS: 10
HOMEBREW_SORBET_RUNTIME: set
Homebrew Ruby: 2.6.10 => /System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/bin/ruby
CPU: 10-core 64-bit arm_firestorm_icestorm
Clang: 15.0.0 build 1500
Git: 2.39.3 => /Applications/Xcode-beta.app/Contents/Developer/usr/bin/git
Curl: 8.1.2 => /usr/bin/curl
macOS: 14.1-arm64
CLT: 15.1.0.0.1.1697767488
Xcode: 15.1 => /Applications/Xcode-beta.app/Contents/Developer
Rosetta 2: false
```

### Output of `brew doctor`

```shell
% brew doctor
Your system is ready to brew.
```

### Description of issue

I would like to put together a formula that has a number of perl deps.

I see some examples in the repo, like this:

https://github.com/Homebrew/homebrew-core/blob/master/Formula/p/po4a.rb

Are such formulae with many deps created and maintained all by hand, or is there a tool that puts these together in some fashion (and perhaps makes keeping them up to date easier)?

Thanks!

**Comment by SMillerDev** [comment, marked as answer]
Unfortunately, that's all by hand.

**Comment by alebcay** [comment]
Separately, check out `brew update-perl-resources` from https://github.com/branchvincent/homebrew-tap. I don't think we use this regularly anywhere but maybe it fits the bill.

**Comment by kencu** [comment]
Oh, yeah ... that looks like it's down the path, for sure. Thanks... I'll see if that works.

There are a dozen or so direct perl deps I need, but then those all have their own recursive deps, so the list adds up pretty fast.
